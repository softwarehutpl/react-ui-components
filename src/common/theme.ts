const theme: { [key: string]: { [key: string]: { [key: string]: string } } } = {
  colors: {
    primary: {
      base: '#000000',
      light: '#ffffff',
      dark: '#a8b3bd',
    },
    secondary: {
      base: '#3690e3',
      light: '#75b0e6',
      dark: '#034d91',
    },
    error: {
      base: '#f44336',
      light: '#f56c62',
      dark: '#a1140a',
    },
    warning: {
      base: '#faa72d',
      light: '#ffc36b',
      dark: '#995b00',
    },
    info: {
      base: '#7979d4',
      light: '#9797d1',
      dark: '#4c4cc7',
    },
    success: {
      base: '#4caf50',
      light: '#88c28a',
      dark: '#124f14',
    },
    disabled: {
      base: '#d9d9d9',
      light: '#f0eded',
      dark: '#8c8989',
    },
  },
  // TODO: change it in dedicated task for that (change ease-out for example and add more options)
  transitions: {
    all: {
      fast: 'all 0.5s ease-out',
      mid: 'all 2s ease-out',
      slow: 'all 4s ease-out',
    },
    background: {
      fast: 'background-color 0.5s ease-out',
      mid: 'background-color 2s ease-out',
      slow: 'background-color 4s ease-out',
    },
    fontColor: {
      fast: 'color 0.5s ease-out',
      mid: 'color 2s ease-out',
      slow: 'color 4s ease-out',
    }
  }
};

export default theme;
