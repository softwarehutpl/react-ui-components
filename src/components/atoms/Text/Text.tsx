import React from 'react';
import styled from 'styled-components';
import { COLOR_OPTIONS, TEXT_ALIGNMENT_OPTIONS, FONT_WEIGHT_OPTIONS } from '../../../common/constants/consts';

interface IText {
  className?: string;
  children: string;
  color?: string;
  fontColor?: string;
  fontWeight?:
    | 'normal'
    | 'bold'
    | '100'
    | '200'
    | '300'
    | '400'
    | '500'
    | '600'
    | '700'
    | '800'
    | '900';
  lineHeight?: number;
  fontSize?: number;
  textAlign?: 'left' | 'right' | 'center' | 'justify';
}

const Text = ({ className, children }: IText) => <p className={className}>{children}</p>;

const defaultProps = {
  color: COLOR_OPTIONS.primary,
  fontSize: 16,
  fontWeight: FONT_WEIGHT_OPTIONS.normal,
  lineHeight: 1.5,
  textAlign: TEXT_ALIGNMENT_OPTIONS.justify,
};

Text.defaultProps = defaultProps;

export default styled(Text)<IText>`
  ${({
    theme,
    color = defaultProps.color,
    fontColor,
    fontSize = defaultProps.fontSize,
    fontWeight = defaultProps.fontWeight,
    lineHeight = defaultProps.lineHeight,
    textAlign = defaultProps.textAlign,
  }) => ({
    color: fontColor || theme.colors[color].base,
    'font-size': `${fontSize}px`,
    'line-height': `${lineHeight}`,
    'font-weight': fontWeight,
    'text-align': textAlign,
  })}
`;
