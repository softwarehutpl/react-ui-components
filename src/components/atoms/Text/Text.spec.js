import React from 'react';
import { create } from 'react-test-renderer';
import 'jest-styled-components';
import { ThemeProvider } from 'styled-components';
import theme from '../../../common/theme';
import Text from './Text';

describe('Text component', () => {
  it('should render correctly', () => {
    const component = create(
      <ThemeProvider theme={theme}>
        <Text>
          some text
        </Text>
      </ThemeProvider>
    ).toJSON();
    expect(component).toMatchSnapshot()
  })

  it('should render with custom class', () => {
    const component = create(
      <ThemeProvider theme={theme}>
        <Text className="test-class">
          some text
        </Text>
      </ThemeProvider>
    ).toJSON();
    expect(component.props.className).toEqual(expect.stringContaining('test-class'));
  });

  it('should render with text', () => {
    const component = create(
      <ThemeProvider theme={theme}>
        <Text>
          foo
        </Text>
      </ThemeProvider>
    ).toJSON();
    expect(component.children[0]).toEqual('foo');
  })

  it('should render text with default style for secondary theme', () => {
    const component = create(
      <ThemeProvider theme={theme}>
        <Text color="secondary">
          foo
        </Text>
      </ThemeProvider>
    ).toJSON();
    expect(component).toHaveStyleRule('color', '#2196f3');
    expect(component).toHaveStyleRule('font-weight', 'normal');
    expect(component).toHaveStyleRule('line-height', '1.5');
    expect(component).toHaveStyleRule('font-size', '16px');
    expect(component).toHaveStyleRule('text-align', 'justify');
  })

  it('should render text with custom style', () => {
    const component = create(
      <ThemeProvider theme={theme}>
        <Text fontColor="red" fontWeight="700" lineHeight={2} fontSize={20} textAlign="center">
          foo
        </Text>
      </ThemeProvider>
    ).toJSON();
    expect(component).toHaveStyleRule('color', 'red');
    expect(component).toHaveStyleRule('font-weight', '700');
    expect(component).toHaveStyleRule('line-height', '2');
    expect(component).toHaveStyleRule('font-size', '20px');
    expect(component).toHaveStyleRule('text-align', 'center');
  })
});
