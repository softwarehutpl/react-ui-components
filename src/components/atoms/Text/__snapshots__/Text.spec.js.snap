// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text component should render correctly 1`] = `
.c0 {
  color: #000000;
  font-size: 16px;
  line-height: 1.5;
  font-weight: normal;
  text-align: justify;
}

<p
  className="c0"
>
  some text
</p>
`;
